Tests/SwiftLintFrameworkTests/ConfigurationTests.swift

```diff
@@ -195,6 +195,22 @@ class ConfigurationTests: XCTestCase {
                           projectMockConfig0CustomPath.configuration(for: file))
     }
 
+    func testConfigurationWithSwiftFileAsRoot() {
+        let configuration = Configuration(path: projectMockYAML0,
+                                          rootPath: projectMockSwift0,
+                                          optional: false, quiet: true)
+        let file = File(path: projectMockSwift0)!
+        XCTAssertEqual(configuration.configuration(for: file), configuration)
+    }
+
+    func testConfigurationWithSwiftFileAsRootAndCustomConfiguration() {
+        let configuration = Configuration(path: projectMockYAML0CustomPath,
+                                          rootPath: projectMockSwift0,
+                                          optional: false, quiet: true)
+        let file = File(path: projectMockSwift0)!
+        XCTAssertEqual(configuration.configuration(for: file), configuration)
```

Comment:
This happens locally on my Mac. I'm pretty sure it's not a bug of comparison, but a consequence of the bug.